### DISCUSSION TITLE
Eclipse seems to be unable to find a type in the same JDK

### DISCUSSION THREAD

**Discussion by guw** [eclipse-jdt/eclipse.jdt.core#2422] [Q&A] [answered]
I am not sure how to reproduce this. Please have a look at the screenshot and the error message reported there.

<img width="2560" alt="image" src="https://github.com/eclipse-jdt/eclipse.jdt.core/assets/191472/adbd4983-f6d6-4c8f-85c8-ae6afd506173">

Is this a broken index or is something else corrupt within the JDK?

> Eclipse IDE for Java Developers (includes Incubating components)
>
> Version: 2024-03 (4.31.0)
> Build id: 20240307-1437

**Comment by iloveeclipse** [comment]
I would assume .claspath is mis-configured. Probably side effect of https://github.com/eclipse-jdt/eclipse.jdt.core/discussions/2421 where some libraries manage to contain "duplicates" of `java.*` packages.

**Reply by guw** [reply]
I thought of this too. However, this is a different project and classpaths are not connected. Also, it does not have org.osgi.foundation jar.

**Comment by guw** [comment, marked as answer]
I managed to make the error go away using the following steps:
* Right click project, select _Build Path > Configure Build Path ..._
* Switch JRE Library to execution environment _Java SE-17_ (which is a different JDK in my case)
-> at this point the problem went away
* Switch JRE Library back to alternate JDK and the one that was originally used
-> at this point the problem did not come back

Thus, it must have been some internal state messed up. 🤷

**Reply by andreasdc** [reply]
> I managed to make the error go away using the following steps:
> 
> * Right click project, select _Build Path > Configure Build Path ..._
> * Switch JRE Library to execution environment _Java SE-17_ (which is a different JDK in my case)
>   -> at this point the problem went away
> * Switch JRE Library back to alternate JDK and the one that was originally used
>   -> at this point the problem did not come back
> 
> Thus, it must have been some internal state messed up. 🤷

It will probably happen again after some time.

**Comment by jukzi** [comment]
probably duplicate of https://github.com/eclipse-jdt/eclipse.jdt.core/issues/2877
